config/config.go

```diff
@@ -92,6 +92,9 @@ func mergeFlagsWithConfig(flags *GlobalFlagValues, config *Config) error {
 	if flags.Token != "" {
 		config.Hazelcast.Cluster.Cloud.Token = strings.TrimSpace(flags.Token)
 		config.Hazelcast.Cluster.Cloud.Enabled = true
+		if config.SSL.ServerName == "" && !config.SSL.InsecureSkipVerify {
```

Comment:
I think it's OK to not check for `InsecureSkipVerify`. Setting server name shouldn't affect that one.